diff --git a/cli/build.rs b/cli/build.rs
index d3f428c50..e869f97ec 100644
--- a/cli/build.rs
+++ b/cli/build.rs
@@ -467,6 +467,22 @@ fn main() {
     ));
     res.compile().unwrap();
   }
+
+  // recover android snapshot
+  if target == "aarch64-linux-android" {
+    let snapshot_dir = PathBuf::from(env::var("GITHUB_WORKSPACE").unwrap()).join("assets/deno_snapshot");
+    let src_compiler_snapshot_path = snapshot_dir.join("COMPILER_SNAPSHOT.bin");
+    let src_cli_snapshot_path = snapshot_dir.join("CLI_SNAPSHOT.bin");
+    let dst_compiler_snapshot_path = o.join("COMPILER_SNAPSHOT.bin");
+    let dst_cli_snapshot_path = o.join("CLI_SNAPSHOT.bin");
+
+    println!("cargo:rerun-if-changed={}", src_compiler_snapshot_path.display());
+    println!("recover: {} => {}", src_compiler_snapshot_path.display(), dst_compiler_snapshot_path.display());
+    std::fs::copy(&src_compiler_snapshot_path, &dst_compiler_snapshot_path).unwrap();
+    println!("cargo:rerun-if-changed={}", src_cli_snapshot_path.display());
+    println!("recover: {} => {}", src_cli_snapshot_path.display(), dst_cli_snapshot_path.display());
+    std::fs::copy(&src_cli_snapshot_path, &dst_cli_snapshot_path).unwrap();
+  }
 }
 
 fn deno_webgpu_get_declaration() -> PathBuf {
