diff --git a/ext/webgpu/byow.rs b/ext/webgpu/byow.rs
index 8864e8d2b..747a23c84 100644
--- a/ext/webgpu/byow.rs
+++ b/ext/webgpu/byow.rs
@@ -6,7 +6,7 @@ use deno_core::op2;
 use deno_core::OpState;
 use deno_core::ResourceId;
 use std::ffi::c_void;
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(any(target_os = "linux", target_os = "macos", target_os = "android"))]
 use std::ptr::NonNull;
 
 use crate::surface::WebGpuSurface;
@@ -103,7 +103,7 @@ fn raw_window(
   Ok((win_handle, display_handle))
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "android"))]
 fn raw_window(
   system: &str,
   window: *const c_void,
